### ISSUE TITLE
ArrayStoreException when using spring-boot-1.2.0.M2 on Wildfly 8.1.0

### ISSUE THREAD

**Issue by BigMichi1** [spring-projects/spring-boot#1696]
i upgraded today from 1.2.0.M1 to 1.2.0.M2 in my development version and tried to deploy my app on wildfly 8.1.0. during app start i got this exception:

```
09:56:25,987 INFO  [org.springframework.core.annotation.AnnotationUtils] (MSC service thread 1-4) Failed to introspect annotations on [class org.springframework.boot.autoconfigure.jersey.JerseyAutoConfiguration]: java.lang.ArrayStoreException: sun.reflect.annotation.TypeNotPresentExceptionProxy
09:56:25,988 ERROR [org.jboss.msc.service.fail] (MSC service thread 1-4) MSC000001: Failed to start service jboss.undertow.deployment.default-server.default-host./steamstats: org.jboss.msc.service.StartException in service jboss.undertow.deployment.default-server.default-host./steamstats: Failed to start service
    at org.jboss.msc.service.ServiceControllerImpl$StartTask.run(ServiceControllerImpl.java:1904) [jboss-msc-1.2.2.Final.jar:1.2.2.Final]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [rt.jar:1.8.0_20]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [rt.jar:1.8.0_20]
    at java.lang.Thread.run(Thread.java:745) [rt.jar:1.8.0_20]
Caused by: java.lang.RuntimeException: java.lang.ArrayStoreException: sun.reflect.annotation.TypeNotPresentExceptionProxy
    at io.undertow.servlet.core.DeploymentManagerImpl.deploy(DeploymentManagerImpl.java:219)
    at org.wildfly.extension.undertow.deployment.UndertowDeploymentService.startContext(UndertowDeploymentService.java:87)
    at org.wildfly.extension.undertow.deployment.UndertowDeploymentService.start(UndertowDeploymentService.java:72)
    at org.jboss.msc.service.ServiceControllerImpl$StartTask.startService(ServiceControllerImpl.java:1948) [jboss-msc-1.2.2.Final.jar:1.2.2.Final]
    at org.jboss.msc.service.ServiceControllerImpl$StartTask.run(ServiceControllerImpl.java:1881) [jboss-msc-1.2.2.Final.jar:1.2.2.Final]
    ... 3 more
Caused by: java.lang.ArrayStoreException: sun.reflect.annotation.TypeNotPresentExceptionProxy
    at sun.reflect.annotation.AnnotationParser.parseClassArray(AnnotationParser.java:724) [rt.jar:1.8.0_20]
    at sun.reflect.annotation.AnnotationParser.parseArray(AnnotationParser.java:531) [rt.jar:1.8.0_20]
    at sun.reflect.annotation.AnnotationParser.parseMemberValue(AnnotationParser.java:355) [rt.jar:1.8.0_20]
    at sun.reflect.annotation.AnnotationParser.parseAnnotation2(AnnotationParser.java:286) [rt.jar:1.8.0_20]
    at sun.reflect.annotation.AnnotationParser.parseAnnotations2(AnnotationParser.java:120) [rt.jar:1.8.0_20]
    at sun.reflect.annotation.AnnotationParser.parseAnnotations(AnnotationParser.java:72) [rt.jar:1.8.0_20]
    at java.lang.Class.createAnnotationData(Class.java:3508) [rt.jar:1.8.0_20]
    at java.lang.Class.annotationData(Class.java:3497) [rt.jar:1.8.0_20]
    at java.lang.Class.getDeclaredAnnotations(Class.java:3464) [rt.jar:1.8.0_20]
    at org.springframework.core.annotation.AnnotationUtils.findAnnotation(AnnotationUtils.java:371)
    at org.springframework.core.annotation.AnnotationUtils.findAnnotation(AnnotationUtils.java:333)
    at org.springframework.core.annotation.OrderUtils.getOrder(OrderUtils.java:58)
    at org.springframework.core.annotation.OrderUtils.getOrder(OrderUtils.java:47)
    at org.springframework.core.annotation.AnnotationAwareOrderComparator.findOrder(AnnotationAwareOrderComparator.java:79)
    at org.springframework.core.OrderComparator.getOrder(OrderComparator.java:105)
    at org.springframework.core.OrderComparator.getOrder(OrderComparator.java:94)
    at org.springframework.core.OrderComparator.doCompare(OrderComparator.java:77)
    at org.springframework.core.OrderComparator.compare(OrderComparator.java:63)
    at java.util.TimSort.countRunAndMakeAscending(TimSort.java:351) [rt.jar:1.8.0_20]
    at java.util.TimSort.sort(TimSort.java:216) [rt.jar:1.8.0_20]
    at java.util.Arrays.sort(Arrays.java:1438) [rt.jar:1.8.0_20]
    at java.util.List.sort(List.java:478) [rt.jar:1.8.0_20]
    at java.util.Collections.sort(Collections.java:175) [rt.jar:1.8.0_20]
    at org.springframework.core.annotation.AnnotationAwareOrderComparator.sort(AnnotationAwareOrderComparator.java:111)
    at org.springframework.web.SpringServletContainerInitializer.onStartup(SpringServletContainerInitializer.java:171)
    at io.undertow.servlet.core.DeploymentManagerImpl.deploy(DeploymentManagerImpl.java:179)
    ... 7 more

09:56:25,997 ERROR [org.jboss.as.controller.management-operation] (XNIO-1 task-3) JBAS014613: Operation ("deploy") failed - address: ([("deployment" => "steamstats.war")]) - failure description: {"JBAS014671: Failed services" => {"jboss.undertow.deployment.default-server.default-host./steamstats" => "org.jboss.msc.service.StartException in service jboss.undertow.deployment.default-server.default-host./steamstats: Failed to start service
    Caused by: java.lang.RuntimeException: java.lang.ArrayStoreException: sun.reflect.annotation.TypeNotPresentExceptionProxy
    Caused by: java.lang.ArrayStoreException: sun.reflect.annotation.TypeNotPresentExceptionProxy"}}
09:56:25,998 ERROR [org.jboss.as.server] (XNIO-1 task-3) JBAS015870: Deploy of deployment "steamstats.war" was rolled back with the following failure message: 
{"JBAS014671: Failed services" => {"jboss.undertow.deployment.default-server.default-host./steamstats" => "org.jboss.msc.service.StartException in service jboss.undertow.deployment.default-server.default-host./steamstats: Failed to start service
    Caused by: java.lang.RuntimeException: java.lang.ArrayStoreException: sun.reflect.annotation.TypeNotPresentExceptionProxy
    Caused by: java.lang.ArrayStoreException: sun.reflect.annotation.TypeNotPresentExceptionProxy"}}
```

1.2.0.M1 works fine here

**Comment by lazee** [comment]
@wilkinsona I think there is a similar case with EndpointWebMvcChildContextConfiguration

```
2015-03-17 17:59:29.112  INFO 99517 --- [           main] o.s.core.annotation.AnnotationUtils      : Failed to introspect annotations on [class org.springframework.boot.actuate.autoconfigure.EndpointWebMvcChildContextConfiguration$EndpointHandlerMappingConfiguration$$EnhancerBySpringCGLIB$$80ee83ce]: java.lang.ArrayStoreException: sun.reflect.annotation.TypeNotPresentExceptionProxy
2015-03-17 17:59:29.113  INFO 99517 --- [           main] o.s.core.annotation.AnnotationUtils      : Failed to introspect annotations on [class org.springframework.boot.actuate.autoconfigure.EndpointWebMvcChildContextConfiguration$EndpointHandlerMappingConfiguration$$EnhancerBySpringCGLIB$$80ee83ce]: java.lang.ArrayStoreException: sun.reflect.annotation.TypeNotPresentExceptionProxy
```

When Spring Security is not in classpath. Should I open an issue on this?

**Comment by wilkinsona** [comment]
@lazee Yes, please. `EndpointWebMvcChildContextConfiguration` isn't a `WebApplicationInitializer` so it's a different (and currently not obvious) cause. A sample app that reproduces the problem would be very helpful.

**Comment by lazee** [comment]
Created #2674

**Comment by wilkinsona** [comment]
Thanks

**Comment by rupams2002** [comment]
I am getting this :
java.lang.ArrayStoreException: sun.reflect.annotation.TypeNotPresentExceptionProxy

**Comment by philwebb** [comment]
@rupams2002 This issue has been closed for a while. If you think you've found a bug please open a new issue with as much information as possible and a project that reproduces it.

**Comment by valters** [comment]
If you are seeing this error (ArrayStoreException: sun.reflect.annotation.TypeNotPresentExceptionProxy) when running tests with surefire, a workaround is discussed at #6254

**Comment by AngelaSunny** [comment]
I also meet this question, becasue someone of the maven dependency(nested)  is conflict with your spring-boot, so you can find the which depency is connect with spring ,so you can exclude it and try run again.

**Comment by asarkar** [comment]
@wilkinsona I have this issue using Boot 2.0 and Spring Security in a project. I understand this issue is closed, but I'm interested in knowing how did you pinpoint the `JerseyAutoConfiguration` based on the Gradle dependencies log. I'm sure understanding that would help others later, as the exception is as close it to useless as it gets: Otherwise, it's always going to be each ticket for itself (teach a man to fish etc).

**Comment by wilkinsona** [comment]
Debug the app with an entry breakpoint on `sun.reflect.annotation.TypeNotPresentExceptionProxy.TypeNotPresentExceptionProxy(String, Throwable)`. The `Throwable` passed into that method is the underlying failure that the JVM swallows.

**Comment by jhawkesworth** [comment]
@wilkinsona - thanks for this tip, you have unblocked my project!
